src/includes/class-alma-wc-payment-validator.php

```diff
@@ -38,47 +38,71 @@ public static function validate_payment( $payment_id ) {
 		if ( $order->get_wc_order()->has_status( apply_filters( 'alma_wc_valid_order_statuses_for_payment_complete', array( 'on-hold', 'pending', 'failed', 'cancelled' ) ) ) ) {
 			if ( $order->get_total() !== $payment->purchase_amount ) {
-				$error = "Order {$order->get_id()} total ({$order->get_total()}) does not match purchase amount of '$payment_id' ($payment->purchase_amount)";
-				$logger->error( $error );
+				$logger->error( 'The total order amount does not match the purchase amount of the payment.', [
+                    'Method' => __METHOD__,
+                    'OrderId' => $order->get_id(),
+                    'OrderAmount' => $order->get_total(),
+                    'PaymentId' => $payment_id,
+                    'PaymentPurchaseAmount' => $payment->purchase_amount
+                ]);
 
 				try {
-					$logger->warning( "Amount mismatch for order {$order->get_id()}" );
 					$alma->payments->flagAsPotentialFraud( $payment_id, Payment::FRAUD_AMOUNT_MISMATCH );
 				} catch ( RequestError $e ) {
-					$logger->warning( $e->getMessage() );
+					$logger->error( 'FlagAsPotentialFraud request error (total amount issue).', [
+                        'Method' => __METHOD__,
+                        'ExceptionMessage'  => $e->getMessage()
+                    ]);
 				}
 
-				throw new Alma_WC_Payment_Validation_Error( $error );
+				throw new Alma_WC_Payment_Validation_Error(sprintf("Order %s total (%s) does not match purchase amount of '%s' (%s)", $order->get_id(), $order->get_total(), $payment_id, $payment->purchase_amount));
 			}
 
 			$first_instalment = $payment->payment_plan[0];
 			if ( ! in_array( $payment->state, array( Payment::STATE_IN_PROGRESS, Payment::STATE_PAID ), true ) ||
 				Instalment::STATE_PAID !== $first_instalment->state ) {
 
-				$error = "Payment '$payment_id': state incorrect $payment->state & $first_instalment->state";
-				$logger->error( $error );
+				$logger->error('State payment error/incorrect.', [
+                    'Method' => __METHOD__,
+                    'OrderId' => $order->get_id(),
+                    'PaymentId' => $payment_id,
+                    'PaymentState' => $payment->state,
+                    'FirstInstalmentState' => $first_instalment->state
+                ]);
 
 				try {
-					$logger->warning( "Payment state error for order {$order->get_id()}" );
 					$alma->payments->flagAsPotentialFraud( $payment_id, Payment::FRAUD_STATE_ERROR );
 				} catch ( RequestError $e ) {
-					$logger->warning( $e->getMessage() );
+                    $logger->error( 'FlagAsPotentialFraud request error (state payment incorrect).', [
+                        'Method' => __METHOD__,
+                        'ExceptionMessage'  => $e->getMessage()
+                   ]);
 				}
 
-				throw new Alma_WC_Payment_Validation_Error( $error );
+				throw new Alma_WC_Payment_Validation_Error(sprintf("Payment '%s': state incorrect %s & %s", $payment_id, $payment->state, $first_instalment->state));
```

Comment:
need translation ?

Reply:
Yes, the error is catch and send back to the front